src/result.cpp

```
@@ -95,4 +186,434 @@ std::map<Label, std::vector<std::string>> AnalysisResult::find_unreachable(const
+
+        // Seed relevant registers from the failed assertion.
+        // Forward analysis records only the first assertion that fails at this label,
+        // so we prefer to seed relevance from that single assertion to keep slices tight.
+        // However, if the first assertion has no register dependencies, we fallback
+        // to all assertions at this label to ensure we produce a usable slice.
+        RelevantState initial_relevance;
+        const auto& assertions = prog.assertions_at(label);
+        if (!assertions.empty()) {
+            // Try the first assertion (the failing one)
+            for (const auto& reg : extract_assertion_registers(assertions[0])) {
+                initial_relevance.registers.insert(reg);
```

Cppcheck (2.19.0)

[style] The function 'update_edge' is never used.

(unusedFunction)